packages/zipkin-instrumentation-koa/src/koaMiddleware.js

```diff
@@ -0,0 +1,28 @@
+const {option: {Some, None}, Instrumentation} = require('zipkin');
+
+module.exports = function zipkinMiddleware({tracer, serviceName = 'unknown', port = 0}) {
+  const instrumentation = new Instrumentation.HttpServer({tracer, serviceName, port});
+
+  return async function zipkinKoaMiddleware(ctx, next) {
+    function readHeader(header) {
+      const val = ctx.request.headers[header.toLowerCase()];
+      if (val != null) {
+        return new Some(val);
+      } else {
+        return None;
+      }
+    }
+    tracer.scoped(async() => {
```

Comment:
usually we use promise style .then() etc. any reason for not doing that in this?

Reply:
No reason, using Promise style also makes sense for me. Will change it.